nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch.java

```diff
@@ -0,0 +1,329 @@
+})
+@InputRequirement(InputRequirement.Requirement.INPUT_ALLOWED)
+@EventDriven
+@Tags({"elasticsearch", "elasticsearch 5", "query", "read", "get", "json"})
+@CapabilityDescription("A processor that allows the user to run a query (with aggregations) written with the " +
+        "ElasticSearch JSON DSL. It currently does not support pagination.")
+public class JsonQueryElasticsearch extends AbstractProcessor {
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder().name("original")
+            .description("All original flowfiles that don't cause an error to occur go to this relationship. " +
+                    "This applies even if you select the \"split up hits\" option to send individual hits to the " +
+                    "\"hits\" relationship.").build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("All FlowFiles that cannot be read from Elasticsearch are routed to this relationship").build();
+
+    public static final Relationship REL_HITS = new Relationship.Builder().name("hits")
+            .description("Search hits are routed to this relationship.")
+            .build();
+
+    public static final Relationship REL_AGGREGATIONS = new Relationship.Builder().name("aggregations")
+            .description("Aggregations are routed to this relationship.")
+            .build();
+
+    public static final PropertyDescriptor INDEX = new PropertyDescriptor.Builder()
+            .name("el-rest-fetch-index")
+            .displayName("Index")
+            .description("The name of the index to read from")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TYPE = new PropertyDescriptor.Builder()
+            .name("el-rest-type")
+            .displayName("Type")
+            .description("The type of this document (used by Elasticsearch for indexing and searching)")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
+            .name("el-rest-query")
+            .displayName("Query")
+            .description("A query in JSON syntax, not Lucene syntax. Ex: " +
+                    "{\n" +
+                    "\t\"query\": {\n" +
+                    "\t\t\"match\": {\n" +
+                    "\t\t\t\"name\": \"John Smith\"\n" +
+                    "\t\t}\n" +
+                    "\t}\n" +
+                    "}")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor QUERY_ATTRIBUTE = new PropertyDescriptor.Builder()
```

Comment:
Where is this prop used? I only see it referenced when it's added to the descriptors. I do see the name named attribute used a lot in the mongo processors though.

Reply:
Fixed.